go/vt/vtgate/evalengine/expressions.go

```diff
@@ -520,6 +520,9 @@ func (t TupleExpr) typeof(*ExpressionEnv) (sqltypes.Type, flag) {
 }
 
 func (c *Column) typeof(env *ExpressionEnv) (sqltypes.Type, flag) {
+	if len(env.Row) <= c.Offset {
+		return sqltypes.Null, 0
```

Comment:
This seems fine, but you need to return the `flagNull|flagNullable` together with the `Null` value, otherwise other expressions will fail to typecheck.